src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/SubmissionManagerImpl.java

```diff
@@ -72,27 +93,32 @@ public SubmissionManagerImpl(SubmissionDispatcher submissionDispatcher) {
      *
      * @param response Response containing a list of tasks.
      * @return List of tasks.
-     * @throws IOException
-     * @throws JSONException
      */
-    private List<ImageTask> extractTasksList(Representation response) throws IOException, JSONException {
+    private List<ImageTask> extractTasksList(Representation response) {
         List<ImageTask> tasks = new ArrayList<>();
-        JsonConverter jsonConverter = new JsonConverter();
-        JSONObject responseJson = jsonConverter.toObject(response, JSONObject.class, null);
-        JSONArray tasksJsonArray = responseJson.getJSONArray("result");
-        for (int i = 0; i < tasksJsonArray.length(); i++) {
-            tasks.add(new ImageTask(tasksJsonArray.optJSONObject(i)));
+        try {
+            JsonConverter jsonConverter = new JsonConverter();
+            JSONObject responseJson = jsonConverter.toObject(response, JSONObject.class, null);
+            JSONArray tasksJsonArray = responseJson.getJSONArray("result");
+            for (int i = 0; i < tasksJsonArray.length(); i++) {
+                tasks.add(new ImageTask(tasksJsonArray.optJSONObject(i)));
+            }
+        } catch (JSONException | IOException e) {
+            LOGGER.error("Error while extracting tasks from response", e);
         }
         return tasks;
     }
 
     /**
-     * Returns the URL of a SAPS remote instance.
+     * Returns a list of URLs of all SAPS neighbors of this instance.
      *
-     * @return URL of a SAPS remote instance.
+     * @return list of URLs of all SAPS neighbors of this instance.
      */
-    private String getRemoteInstanceUrl() {
-        return REMOTE_INSTANCE_URL;
+    private String[] getSAPSNeighborsUrls() {
+        String separator = ";";
+        String SAPSNeighborsUrls = properties.getProperty(SAPS_NEIGHBORS_URLS);
+        return !Objects.isNull(SAPSNeighborsUrls) ? SAPSNeighborsUrls.split(separator)
```

Comment:
maybe, it a matter of code style ... Objects.nonNull sounds better

Reply:
It really sounds, I just didn't know this method exists. I'll change